### ISSUE TITLE
Inability to place structures sufficiently far from colony guarantees an eventual loss

### ISSUE THREAD

**Issue by Inglonias** [OutpostUniverse/OPHD#1267]
#1223 has been marked "Won't fix", which is fine. The original Outpost game prevented players from building too far from their colony as well. That said, something does need to change in this regard. There are two underlying issues that need to be fixed:

* Nothing in the UI tells the player that this is a limitation when they begin the game. Displaying a color-coded radius around the SEED Lander before it lands will resolve this problem. Two colors for the overlay are probably needed. The first is the maximum distance that the player will be allowed to place structures, and the second is maximum communications range assuming the first circumference is covered in comm towers. These two pieces of info are required to determine the really important part - How many of the planet's mining sites are available to your colony.
* This limitation places an overall cap on the amount of mineral resources available to your colony over the course of the game. This isn't a problem in and of itself, but the game needs to be designed to account for this. There needs to be enough available minerals to make the colony self-sustaining long enough to win the game, or if there isn't going to be a final victory condition, long enough to research the tech needed to get past this limitation (MPG, a portal to the mineral dimension, whatever)

**Comment by ldicker83** [comment]
Robots can be placed anywhere on the surface so long as they are within comm range (this includes mining robots which extends to mine facilities). Comm towers are excluded from this limitation of structure placement as should any structure that is considered self-contained (need to check the logic to make sure that's the case and enable that if it isn't)

I want to eventually extend building placement to anywhere within comm range. There are some implications to the underlying logic that come from this, namely what happens if comm range breaks?

I've seen a couple of save games where comm towers are built to extend comm range for mine facilities only to be removed afterward to save on power consumption. Great use of resources but a bit of an exploit of game mechanics and not intended for gameplay.

Also, I wanted to add the concept of 'colony sprawl' -- if structures are too far away things are going to take awhile to get there, CHAP facilities may need a boost, etc. Something I want to think about before just blindly extending the range.

So ultimately not a concept I want to outright deny, but in the original issue it doesn't account for the issues I'm thinking about atm.

**Comment by Inglonias** [comment]
Oh. I didn't know comm towers were exempt. That does change things a bit.

**Comment by ldicker83** [comment]
Yup -- any structure that's considered 'self contained' such as a comm tower is exempt from this. I know the mine facilities, roads and tubes are also considered self contained so should be affected.

**Comment by DanRStevens** [comment]
Given that comm towers allow distant mines to be used, and we've since updated ore haul calculations to decrease the drop off in ore haul rates for distant mines, I think we can claim this issue has been addressed.

Related:
- Issue #1846

There was also an idea discussed on Discord, where maybe the `CommandCenter` should allow a certain number of structures to be controlled (assuming tube connection), rather than limit it by distance. That would be another way to handle distant mines, and perhaps allow building smelters closer to distant mines.

If there's something I've missed, or more thoughts to add, we can always reopen this.
